config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1640,6 +1687,9 @@ else if (this.securityContextRepository != null) {
 		if (this.logout != null) {
 			this.logout.configure(this);
 		}
+		if (this.oneTimeTokenLogin != null) {
+			this.oneTimeTokenLogin.configure(this);
```

Comment:
This will need to include negotiating the security context repository (see the way that `formLogin` and `oauth2Login` are configured in this method). To achieve this, `OneTimeTokenLoginSpec` needs a `securityContextRepository` method.

Because `AuthenticationWebFilter` defaults to a no-op security context repository, I believe this contributes to why the user is not ultimately logged in after posting the token.